doc/cascadia/Keybindings-Arguments.md

```diff
@@ -0,0 +1,315 @@
+`ShortcutAction`.
+
+Then, in `App`, we'll handle each of these events. We set up lambdas as event
+handlers for each event in `App::_HookupKeyBindings`. In each of those
+functions, We'll inspect the `IKeyBindingArgs` parameter, and use args from its
+implementation to call callbacks in the `App` class. We will update `App` to
+have methods defined with the actual keybinding function signatures.
+
+Instead of:
+
+```c++
+    void App::_HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept
+    {
+        // ...
+        bindings.NewTabWithProfile([this](const auto index) { _OpenNewTab({ index }); });
+    }
+```
+
+The code will look like:
+
+```c++
+    void App::_HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept
+    {
+        // ...
+        bindings.NewTabWithProfile({ this, &App::_OpenNewTab });
+    }
+    // ...
+    void App::_OpenNewTab(const TerminalApp::AppKeyBindings& sender, const NewTabEventArgs& args)
+    {
+        auto profileIndex = args.ProfileIndex();
+        args.Handled(true);
+        // ...
+    }
+```
+
+### Handling Keybinding Events
+
+Commmon to all implementations of `IKeyBindingArgs` is the `Handled` property.
+This will let the app indicate if it was able to actually process a keybinding
+event or not. While in the large majority of cases, the events will all be
+marked handled, there are some scenarios where the Terminal will need to know if
+the event could not be performed. For example, in the case of the `copy` event,
+the Terminal is only capable of copying text if there's actually a selection
+active. If there isn't a selection active, the `App` should make sure to mark
+the event as not handled (`args.Handled(false)`).
+
+When an even is _not_ handled, we'll make sure to return `false` from
+`AppKeyBindings::TryKeyChord`, so that the terminal has another chance to
+actually use that keypress.
+
+### Serializing KeyBinding Arguments
+
+Similar to how we parse arguments from the json, we'll need to update the
+`AppKeyBindingsSerialization` code to be able to serialize the arguments from a
+particular `IKeyBindingArgs`.
+
+## UI/UX Design
+
+### Keybindings in the New Tab Dropdown
+
+Small modifications will need to be made to the code responsible for the new tab
+dropdown. The new tab dropdown currently also displays the keybindings for each
+profile in the new tab dropdown. It does this by querying for the keybinding
+associated with each action. As we'll be removing the old `ShortcutAction`s that
+this dropdown uses, we'll need a new way to find which key chord corresponds to
+opening a given profile.
+
+We'll need to be able to not only lookup a keybinding by `ShortcutAction`, but
+also by a `ShortcutAction` and `IKeyBindingArgs`. We'll need to update the
+`AppKeyBindings::GetKeyBinding` method to also accept a `IKeyBindingArgs`. We'll
+also probably want each `IKeyBindingArgs` implementation to define `operator==`,

```

Comment:
🚨 🚨 🚨 🚨 🚨  operator overloads on winrt things is a biiiig no

Reply:
Good to know, I'll update that